packages/@o3r/core/schematics/ng-add-create/index.ts

```diff
@@ -0,0 +1,74 @@
+import { chain, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getTemplateFolder } from '@o3r/schematics';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import {PackageJson} from 'type-fest';
+
+import { apply, MergeStrategy, mergeWith, renameTemplateFiles, template, url } from '@angular-devkit/schematics';
+import { NgGenerateUpdateSchematicsSchema } from './schema';
+
+/**
+ * Rule factory to include `ng add` skelethon
+ *
+ * @param options
+ * @param rootPath
+ */
+export function updateTemplates(options: NgGenerateUpdateSchematicsSchema, rootPath: string): Rule {
+
+  return (tree: Tree, context: SchematicContext) => {
+
+    // register scripts
+    if (tree.exists('/package.json')) {
+      const packageJson: PackageJson = JSON.parse(tree.read('/package.json')!.toString());
+      if (!packageJson.devDependencies?.['@angular/compiler']) {
+        context.logger.info('You are not in an angular project. Ng add skelethon creation aborted.');
```

Comment:
The message is not clear, you could be running in Angular app but not having explicitly `@angular/compiler` in your devDeps.

Reply:
I switched it to angular/cli as this is added as as a dev dependency when we run the ng new generator.
What I wanted was to abort the everything if I am not in an angular project. 
Do you think we can still add the schematics , and eventually put the 'latest' version for '@angular-devkit/schematics' if we don;t find the angular cli in the package.json?

Reply:
The thing is that it could be on `dep` or `devDep` (for application it is quite the same).
I think the message should be clear about what it is actually required (and maybe to check both `dep` and `devDep`